### PR TITLE
830

### PR DESCRIPTION
**Pull request by ivicac**
- ** 0 - client - Update dependencies**
- **0 - client - Update shadcn**
- **0 - client - Update titles**
- **0 - client - Fix Authorization drop-down in ConnectionDialog**
- **377 - client - Add WorkflowExecutionsTestOutput for showing the output of workflow test executions in the workflow visual/code editors**
- **0 - client - Update dependencies**
- **830 - client - Update ProjectInstanceDialog**
- **830 - client - SF**
- **0 - client - Rename**
- **0 - client - Fix project status**
- **0 - client - Update property**
- **0 - client - Make the x button smaller in the tag list**
- **0 - client - Update button labels**
- **520 - client - Reorganize LeftSidebarNav**
- **0 - client - Only workflow execution with FAILED status should use red color**

> Pull Request Template
>
> Use this template to quickly create a well written pull request. Delete all quotes before creating the pull request.

## Description

> Please include a summary of the changes and which issue has been fixed. Please also include relevant motivation
> and context. List any dependencies that are required for this change.

Fixes # (issue)

> if no issue exists, please create an issue and ask the maintainers about this first

## Type of change

> Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- New connector (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- This change requires a documentation update

## How Has This Been Tested?

> Please describe the tests that you ran to verify your changes. Provide instructions, so we can reproduce.
> Please also list any relevant details for your test configuration.

- Test A
- Test B

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: client/src/pages/automation/project/Project.tsx [hunk 1]
```diff
@@ -641,12 +630,32 @@ const Project = () => {
...
+                                    componentDefinitions={componentDefinitions}
+                                    projectId={+projectId!}
+                                    taskDispatcherDefinitions={taskDispatcherDefinitions}
+                                    updateWorkflowMutation={updateWorkflowMutation}
+                                />
+                            </ResizablePanel>
+
+                            <ResizableHandle />
+
+                            <ResizablePanel className="bg-white" defaultSize={0} ref={bottomImperativePanelHandleRef}>
```
Comment (kresimir-coko) [inline client/src/pages/automation/project/Project.tsx]: Ne sviđa mi se ovo imenovanje - `bottomResizablePanelRef` je puno bolje, imperative/declarative nije nešto šta se specificira u imenu (koliko ja znam), bez `Handle` jer nije postavljen `ref ` na `handle` nego na čitav `panel`

File: client/src/pages/automation/project/Project.tsx [hunk 2]
```diff
@@ -381,33 +399,7 @@ const Project = () => {
...
-                                    <RefreshCwIcon className="size-5" />
-                                </span>
-
-                                <span className="text-muted-foreground">Workflow is running...</span>
-                            </div>
-                        )}
-                    </>
-                }
-                leftSidebarOpen={leftSidebarOpen}
+                leftSidebarOpen={false}
```
Comment (kresimir-coko) [inline client/src/pages/automation/project/Project.tsx:399]: Ako izostaviš ovaj prop, trebalo bi po defaultu bit `false` - boolean propovi su `false` osim ako ih ne definiraš (i to ne triba `={value}` dio nego samo `leftSidebarOpen`)

File: client/src/pages/automation/project/Project.tsx [hunk 3]
```diff
@@ -551,87 +537,90 @@ const Project = () => {
...
                                         </span>
                                     </TooltipTrigger>
 
-                                    <TooltipContent>The project is published</TooltipContent>
+                                    <TooltipContent>
+                                        {project?.publishedDate ? `The project is published` : `Publish the project.`}
+                                    </TooltipContent>
                                 </Tooltip>
-                            )}
+                            </div>
```
Comment (kresimir-coko) [inline client/src/pages/automation/project/Project.tsx]: Mislin da je ovaj `div` višak

Comment (kresimir-coko) [reply client/src/pages/automation/project/Project.tsx]: Kao i `span` oko `Button`-a, ako nije višak onda je semantički krivo

File: client/src/pages/automation/project/components/WorkflowExecutionsTestOutput.tsx [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+    const job = workflowTestExecution?.job as JobModel;
+    const triggerExecution = workflowTestExecution?.triggerExecution as TriggerExecutionModel;
+
+    return (
+        <div className="flex size-full flex-col">
+            <div className="flex items-center justify-between border-b-2 border-b-muted px-3 py-1">
+                {workflowTestExecution ? (
+                    <WorkflowExecutionsTestOutputHeader job={job} triggerExecution={triggerExecution} />
+                ) : (
+                    <div className="text-sm uppercase">Test Output</div>
```
Comment (kresimir-coko) [inline client/src/pages/automation/project/components/WorkflowExecutionsTestOutput.tsx]: `span`

File: client/src/pages/automation/project/components/WorkflowExecutionsTestOutput.tsx [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+                )}
+
+                {onCloseClick && (
+                    <div className="cursor-pointer p-2" onClick={() => onCloseClick()}>
+                        <ChevronDownIcon className="h-5" />
+                    </div>
```
Comment (kresimir-coko) [inline client/src/pages/automation/project/components/WorkflowExecutionsTestOutput.tsx]: Idealno bi bilo kad bi bija `Button`

File: client/src/pages/automation/project/components/WorkflowExecutionsTestOutput.tsx [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+                                <WorkflowExecutionContent {...content} />
+                            </ResizablePanel>
+                        </ResizablePanelGroup>
+                    ) : (
+                        <div className="flex size-full items-center justify-center gap-x-1 p-3 text-muted-foreground">
+                            <RefreshCwOffIcon className="size-5" />
+
+                            <span>The workflow has not yet been executed.</span>
+                        </div>
+                    )}
```
Comment (kresimir-coko) [inline client/src/pages/automation/project/components/WorkflowExecutionsTestOutput.tsx]: Pošto ode imamo conditional operator na 2 razine za renderanje, ja bi ga razbija u ifove

```tsx
{workflowIsRunning && (...)}

{(!workflowIsRunning && workflowTestExecution?.job) ? (...) : (...)}
```

File: client/src/pages/automation/project/components/WorkflowNodesTabs.tsx [hunk 7]
```diff
@@ -53,71 +53,79 @@ const WorkflowNodesTabs = ({
...
-                                    (componentDefinition: ComponentDefinitionBasicModel) => (
-                                        <WorkflowNodesTabsItem
-                                            draggable={itemsDraggable}
-                                            handleClick={() => onItemClick && onItemClick(componentDefinition)}
-                                            key={componentDefinition.name}
-                                            node={componentDefinition}
-                                        />
+            <div className="relative flex flex-1">
+                <div
+                    className={twMerge('overflow-y-auto flex absolute', popover && 'h-80 w-full', !popover && 'h-full')}
```
Comment (kresimir-coko) [inline client/src/pages/automation/project/components/WorkflowNodesTabs.tsx:58]: mislin da ne triba ovaj `absolute`, ali zapisujem sebi da provjerin sve, baš je loša praksa

File: client/src/pages/platform/workflow-executions/components/WorkflowExecutionContent.tsx [hunk 8]
```diff
@@ -63,74 +62,76 @@ const WorkflowExecutionDetailsAccordionContent = ({
...
-                                    {output ? (
-                                        typeof output === 'object' ? (
-                                            <ReactJson
-                                                collapsed={false}
-                                                enableClipboard={false}
-                                                src={output as object}
-                                            />
+                                    <div className="max-h-[80vh] overflow-y-auto">
+                                        {output ? (
+                                            typeof output === 'object' ? (
```
Comment (kresimir-coko) [inline client/src/pages/platform/workflow-executions/components/WorkflowExecutionContent.tsx:104]: pazi se ovog `typeof x === 'object'` - puno toga u JS-u je `object`, pa i `array`

File: client/src/pages/platform/workflow-executions/components/WorkflowExecutionContent.tsx [hunk 9]
```diff
@@ -63,74 +62,76 @@ const WorkflowExecutionDetailsAccordionContent = ({
...
+                                                <ReactJson
+                                                    collapsed={false}
+                                                    enableClipboard={false}
```
Comment (kresimir-coko) [inline client/src/pages/platform/workflow-executions/components/WorkflowExecutionContent.tsx:107]: `false` propove je vjerojatno dovoljno izostavit - ovisi o implementaciji komponente, ali tako inače funkcioniraju

**Review by kresimir-coko — CHANGES_REQUESTED** [review]

File: client/src/pages/automation/project/Project.tsx (hunk 2 above)
Comment (ivicac) [reply client/src/pages/automation/project/Project.tsx:399]: default je true :

**Pull request merged**